types/castv2/index.d.ts

```diff
@@ -0,0 +1,119 @@
+/// <reference types="node" />
+
+import { EventEmitter } from "events";
+import { TlsOptions } from "tls";
+
+// To start a new client.
+export interface ClientOptions {
+    host: string;
+    port?: number;
+    rejectUnauthorized?: boolean;
+}
+
+// A general message in either direction.
+export interface Message {
+    type: string;
+}
+
+// A specific connection message sent to the receiver.
+export interface ConnectMessage extends Message {
+    type: "CONNECT";
+}
+
+// A general request sent to the receiver.
+export interface RequestMessage extends Message {
+    requestId: number;
+}
+
+// A specific request sent to the receiver to launch an application.
+export interface LaunchRequestMessage extends RequestMessage {
+    type: "LAUNCH";
+    appId: string;
+    commandParameters?: string;
+}
+
+// A specific request sent to the receiver to stop the current application.
+export interface StopRequestMessage extends RequestMessage {
+    type: "STOP";
+}
+
+// A general response from the receiver.
+export interface ReceiverMessage extends Message {
+    requestId: number;
+}
+
+export interface ApplicationNamespace {
+    name: string;
+}
+
+export interface ApplicationInfo {
+    appId: string;
+    appType: string;
+    displayName: string;
+    iconUrl: string;
+    isIdleScreen: boolean;
+    launchedFromCloud: boolean;
+    namespaces: ApplicationNamespace[];
+    sessionId: string;
+    statusText: string;
+    transportId: string;
+    universalAppId: string;
+}
+
+export interface VolumeInfo {
+    controlType: string;
+    level: number;
+    muted: boolean;
+    stepInterval: number;
+}
+
+// A status message from the receiver.
+export interface ReceiverStatusMessage extends ReceiverMessage {
+    type: "RECEIVER_STATUS";
+    status: {
+        applications?: ApplicationInfo[];
+        volume?: VolumeInfo;
+        isActiveInput?: boolean;
+        isStandBy?: boolean;
+        userEq?: unknown; // Exact format undocumented
+    };
+}
+
+// A launch error message from the receiver.
+export interface ReceiverLaunchErrorMessage extends ReceiverMessage {
+    type: "LAUNCH_ERROR";
+    reason: string;
+}
+
+export interface ReceiverLaunchStatusMessage extends ReceiverMessage {
+    type: "LAUNCH_STATUS";
+    status: "USER_PENDING_AUTHORIZATION" | "USER_ALLOWED";
+}
+
+export type MessageHandler = (data: ReceiverMessage, broadcast: boolean) => void;
+
+export type ErrorHandler = (error: Error) => void;
+
+export class Channel extends EventEmitter {
+    // Do not construct directly.  Use Client's createChannel().
+    private constructor();
+
+    send(data: string | Buffer): void;
+    send<T extends Message>(data: T): void;
+
+    close(): void;
+
+    on(eventName: "message", callback: MessageHandler): this;
+}
+
+export class Client extends EventEmitter {
+    constructor();
+
```

Comment:
```suggestion
```
(every class ships parameterless constructor)

Reply:
I kept it because it was in the output generated by `dts-gen`, but it makes sense to drop it.  Thanks!